packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/features/listFeatures.ts

```diff
@@ -164,27 +164,25 @@ const AutoBullet: BuildInEditFeature<PluginKeyboardEvent> = {
                 let regions: RegionBase[];
                 let searcher = editor.getContentSearcherOfCursor();
                 let textBeforeCursor = searcher.getSubStringBefore(4);
-                let rangeToDelete = searcher.getRangeFromText(
-                    textBeforeCursor,
-                    true /*exactMatch*/
-                );
+                let textRange = searcher.getRangeFromText(textBeforeCursor, true /*exactMatch*/);
 
-                if (!rangeToDelete) {
+                if (!textRange) {
                     // no op if the range can't be found
                 } else if (
                     textBeforeCursor.indexOf('*') == 0 ||
                     textBeforeCursor.indexOf('-') == 0
                 ) {
-                    prepareAutoBullet(editor, rangeToDelete);
+                    prepareAutoBullet(editor, textRange);
                     toggleBullet(editor);
                 } else if (isAListPattern(textBeforeCursor)) {
-                    prepareAutoBullet(editor, rangeToDelete);
+                    prepareAutoBullet(editor, textRange);
                     toggleNumbering(editor);
                 } else if ((regions = editor.getSelectedRegions()) && regions.length == 1) {
                     const num = parseInt(textBeforeCursor);
-                    prepareAutoBullet(editor, rangeToDelete);
+                    prepareAutoBullet(editor, textRange);
                     toggleNumbering(editor, num);
                 }
+                searcher.getRangeFromText(textBeforeCursor, true /*exactMatch*/)?.deleteContents();
```

Comment:
I moved this from the prepareAutoBullet function, because before, we deleted the inline element tag with the styles applied to it, so we did not have styles to use in the List Item. 
So, I moved this after toggling the list type so we maintain the styles in the list.

Reply:
Makes sense. Thanks.